core/src/main/java/com/valtech/aem/saas/core/http/response/FacetFieldsDataExtractionStrategy.java

```diff
@@ -0,0 +1,53 @@
+package com.valtech.aem.saas.core.http.response;
+
+import com.google.gson.JsonElement;
+import com.valtech.aem.saas.api.fulltextsearch.dto.FacetFieldResultDTO;
+import com.valtech.aem.saas.api.fulltextsearch.dto.FacetFieldResultsDTO;
+import com.valtech.aem.saas.core.http.response.dto.FacetCountsDTO;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Optional;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
+import lombok.NonNull;
+import org.apache.commons.collections4.MapUtils;
+
+/**
+ * A strategy for extracting response body data.
+ */
+public final class FacetFieldsDataExtractionStrategy implements
+    SearchResponseDataExtractionStrategy<List<FacetFieldResultsDTO>> {
+
+  @Override
+  public String propertyName() {
+    throw new UnsupportedOperationException();
+  }
+
+  @Override
+  public Optional<List<FacetFieldResultsDTO>> getData(JsonElement response) {
+    return new FacetCountsDataExtractionStrategy().getData(response)
+        .map(this::getFacetResults);
+  }
+
+  private List<FacetFieldResultsDTO> getFacetResults(@NonNull FacetCountsDTO facetCounts) {
+    return Optional.ofNullable(facetCounts.getFacetFields())
+        .filter(MapUtils::isNotEmpty)
+        .map(HashMap::entrySet)
+        .map(Collection::stream)
+        .orElse(Stream.empty())
+        .map(entry -> FacetFieldResultsDTO.builder().fieldName(entry.getKey())
+            .items(getFacetFieldItems(entry.getValue())).build())
+        .collect(Collectors.toList());
+  }
+
+  private List<FacetFieldResultDTO> getFacetFieldItems(@NonNull List<JsonElement> facetFieldsList) {
+    List<FacetFieldResultDTO> items = new ArrayList<>();
+    for (int i = 0; i < facetFieldsList.size(); i = i + 2) {
+      items.add(FacetFieldResultDTO.builder().text(facetFieldsList.get(i).getAsString()).count(
+          facetFieldsList.get(i + 1).getAsBigDecimal().intValue()).build());
```

Comment:
mmh why do we convert first to decimal and then to int?

Reply:
The value received from solr is of decimal format, even though the value is of integer nature. So i need to get it as decimal and convert it to the desired int format.

Reply:
weired, can we doublecheck this with saas team?

Reply:
I just checked the saas api's response, it is an int value there. It seems I was wrong about it. I can use `getAsInt` method instead. 
It could be that I had the issue when I serialized facet fields like **HashMap<String, List<Object>>**
I've recently changed it to **HashMap<String, List<JsonElement>>** and it simplified the parsing